### DISCUSSION TITLE
Identifying all object in HFSS with properties using a given variable

### DISCUSSION THREAD

**Discussion by amichel0205** [ansys/pyaedt#4282] [General]
Hello All,
 
Does any one of you has already looked how to list all object using a specific variable.
e.g. A design variable L1 is used in Xsize property of box 2_pc and 4_pc.
I would like to give variable name L1 as input  and list these two objects 2_pc and 4_pc as output.
 
I've already look about properties from:

aedtapp.modeler.get_object_from_name("2_pc")
 
but even if using variable the bounding_box properties list numerical value and not variable name.
 
Thanks

**Comment by gmalinve** [comment]
Hi @amichel0205,

you should use the history method in a list comprehension:

`[o for o in m3d.modeler.object_list if o.history().props.get("XSize") == "XLength"]`

hope this helps!

**Comment by amichel0205** [comment]
@gmalinve 
Thanks Giulia.
It works as expected.

BR,

Alain
